cluster/sbin/collect-debug-artifacts.sh

```diff
@@ -396,31 +424,42 @@ function collect_on_remote {
   for f in "${files[@]}"
   do
     if [ "${verbose}" = "1" ]; then
-      echo "copying file ${f} in dir ${tmp_dir}"
+      echo "copying file ${f} in dir ${tmp_dir}/"
     fi
     cp $f "${tmp_dir}/"
+    returnval=$?
+    if [ "${verbose}" = "1" ]; then
+      echo "copied and returnval=${returnval}"
+    fi
   done
 
-  if [ "${verbose}" = "1" ]; then
-    echo "FILES=${files[@]} will be rsynced to ${collector_host}:$collector_dir}"
-  fi
+  # gzip all the files so that rsync is fast
+  cd "${tmp_dir}"
+  for i in `ls`
+  do
+    if [ "${verbose}" = "1" ]; then
+      echo "zipping ${i}"
+    fi
+    gzip $i
+    if [ "${verbose}" = "1" ]; then
+      echo "zipp of ${i} done"
+    fi
+  done
 
-  cd "${tmp_dir}/.."
   if [ "${verbose}" = "1" ]; then
-    tar cvzf "${tmp_dir}.tar.gz" $(basename $tmp_dir) && \
-      rsync -av "${tmp_dir}.tar.gz" "${collector_host}:${collector_dir}/" && \
-        rm -f "${tmp_dir}.tar.gz" && rm -rf "${tmp_dir}"
-  else
-    tar czf "${tmp_dir}.tar.gz" $(basename $tmp_dir) && \
-      rsync -a "${tmp_dir}.tar.gz" "${collector_host}:${collector_dir}/" && \
-        rm -f "${tmp_dir}.tar.gz" && rm -rf "${tmp_dir}"
+    echo "FILES=${files[@]} zipped and copied to ${collector_host}:$tmp_dir}"
   fi
 }
 
 check_configs
 # Assuming each line in the members info file has the following format
 # host pid cwd
 
+if [ -n "${TAR_FILE}" ]; then
+  extract "${TAR_FILE}"
```

Comment:
run this in a sub-shell "( extract ... )" since the extract function is doing a "cd" so we don't want user's shell working directory itself to change.